### DISCUSSION TITLE
What about `sphinx.ext.doctest`?

### DISCUSSION THREAD

**Discussion by tony** [executablebooks/meta#776] [Q&A]
So far, nothing comes up when searching for earlier discussions.

Is there support for [`sphinx.ext.doctest`](https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html)?

Is this project the appropriate place to bring it up, or is this out of scope?

I also opened a related question in a [pytest-sphinx](https://github.com/thisch/pytest-sphinx/): https://github.com/thisch/pytest-sphinx/issues/30

**Comment by mforbes** [comment]
I would like to use this too.  Currently, if you use `sphinx.ext.doctest`, then you can add `{doctest}` directives and use them by running `make doctest`.  However, as far as I can tell, this will not included any of the `{code-cells}` from that document - hence you cannot test code you have defined there, and will not be run as part of the generated document.  You must run this independently.

See also https://github.com/executablebooks/MyST-NB/issues/290.
